src/app/features/public-form/public-form-permission.service.spec.ts

```diff
@@ -0,0 +1,189 @@
+import { TestBed } from "@angular/core/testing";
+import { PublicFormPermissionService } from "./public-form-permission.service";
+import { EntityMapperService } from "../../core/entity/entity-mapper/entity-mapper.service";
+import { SessionSubject } from "../../core/session/auth/session-info";
+import { Config } from "../../core/config/config";
+import { BehaviorSubject } from "rxjs";
+
+describe("PublicFormPermissionService", () => {
+  let service: PublicFormPermissionService;
+  let mockEntityMapper: jasmine.SpyObj<EntityMapperService>;
+  let mockSessionSubject: BehaviorSubject<any>;
+
+  beforeEach(() => {
+    mockEntityMapper = jasmine.createSpyObj("EntityMapperService", [
+      "load",
+      "save",
+    ]);
+    mockSessionSubject = new BehaviorSubject(null);
+
+    TestBed.configureTestingModule({
+      providers: [
+        PublicFormPermissionService,
+        { provide: EntityMapperService, useValue: mockEntityMapper },
+        { provide: SessionSubject, useValue: mockSessionSubject },
+      ],
+    });
+    service = TestBed.inject(PublicFormPermissionService);
+  });
+
+  it("should be created", () => {
+    expect(service).toBeTruthy();
+  });
+
+  it("should allow access when permissions config cannot be loaded", async () => {
+    mockEntityMapper.load.and.rejectWith(new Error("Config not found"));
+
+    const result = await service.hasPublicCreatePermission("Child");
+
+    expect(result).toBeTrue();
+  });
+
+  it("should allow access when no permissions are configured", async () => {
```

Comment:
I think in that case it actually doesn't work and needs to create a permission doc ... Do you have a system where you can test if public forms work if no Config:Permission doc exists?